### DISCUSSION TITLE
Logging configuration and overrides

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco-CMS#13356] [Features and ideas]
Hi!

When we're releasing using Octopus deploy each release gets a new folder with the version number eg.
`
Site/1.0.0
Site/1.1.1
`

We want to keep all log-files in one single place so my objective is to configure a different folder to store logs and for the Log Viewer to read logs.

I've noticed that there is an interface `ILoggingConfiguration` that is used eg. by the LogViewer but the path where to write the files is hardcoded in `Umbraco.Web.Common.Extensions.ServiceCollectionExtensions`. Maybe I'm not understanding everything correctly and I also think that the documentation lacks information about how I would do this.

Basically, it would be pretty simple if `ILoggingConfiguration` could also be used to define where the log files should be created/written.

I'm I missing something here? Maybe there is a way to do this but I haven't managed to find it in the docs or the source.

I've look at these docs:
https://our.umbraco.com/documentation/Umbraco-Cloud/Troubleshooting/Log-files/
https://our.umbraco.com/Documentation/Fundamentals/Code/Debugging/Logging/
https://our.umbraco.com/Documentation/Reference/Configuration/Serilog/

And Yes, I guess I could be using a custom sink or something but I just want a simple way to configure another directory to use.

Cheers!

**Comment by bergmania** [comment]
Hi @enkelmedia 

Seems like you are right, the configuration is not really extendible, which it should be. To me, it just looks like something that should be fixed. Maybe even as an up-for-grabs, if you are interested in taking a look :) 

What's important is that the new logging dir is used by both the logger and the log viewer.

**Comment by Zeegaan** [comment]
This is now implemented in https://github.com/umbraco/Umbraco-CMS/pull/13485 🚀

**Comment by Nikkelmann** [comment]
Does this also include the option to disable the UmbracoFile logging or replace it with a custom sink?

**Comment by bergmania** [comment]
You can't disable the UmbracoFile, but you can limit what is logged to only Fatal events. Adding custom sinks have been possible all the time. This is what happens when in Development, where a custom console sink is added.

**Reply by Nikkelmann** [reply]
But it would still create the log files, right?
In the cases where we replace it with our own log sink, there should be no reason to have this build-in file sink.

**Reply by bergmania** [reply]
Currently the log viewer only reads from this file, IIRC it will first be created when the first log entry is created. Having set the minimum logging level high, would lead to it never being read.
